crates/runtime/src/search/rrf.rs

```diff
@@ -159,14 +279,21 @@ impl ReciprocalRankFusion {
         }
     }
 
-    // Given arguments to n search calls: execute searches, generate row IDs, rank by score, JOIN,
-    // then finally re-rank and sort fused results
-    fn rerank_and_fuse_df(&self, args: &ReciprocalRankFusionArgs) -> Result<DataFrame> {
-        let subquery_dfs = self.prepare_and_execute_subqueries(args)?;
-
+    fn compute_score_expr(
```

Comment:
Lifted `score_expr` building into a new function that will apply the relevant boosting params if they're defined